platform-bukkit/src/main/java/net/kyori/adventure/platform/bukkit/MinecraftComponentSerializer.java

```diff
@@ -126,26 +129,45 @@ public static boolean isSupported() {
           if (gsonField != null) {
             gsonField.setAccessible(true);
             gson = gsonField.get(null);
-          } else {
-            final Method[] declaredMethods = chatSerializerClass.getDeclaredMethods();
-            final Method deserialize = Arrays.stream(declaredMethods)
-              .filter(m -> Modifier.isStatic(m.getModifiers()))
-              .filter(m -> CLASS_CHAT_COMPONENT.isAssignableFrom(m.getReturnType()))
-              .filter(m -> m.getParameterCount() == 1 && m.getParameterTypes()[0].equals(String.class))
-              .min(Comparator.comparing(Method::getName)) // prefer the #a method
-              .orElse(null);
-            final Method serialize = Arrays.stream(declaredMethods)
-              .filter(m -> Modifier.isStatic(m.getModifiers()))
-              .filter(m -> m.getReturnType().equals(String.class))
-              .filter(m -> m.getParameterCount() == 1 && CLASS_CHAT_COMPONENT.isAssignableFrom(m.getParameterTypes()[0]))
-              .findFirst()
-              .orElse(null);
-            if (deserialize != null) {
-              textSerializerDeserialize = lookup().unreflect(deserialize);
-            }
-            if (serialize != null) {
-              textSerializerSerialize = lookup().unreflect(serialize);
-            }
+          }
+        }
+        for (final Class<?> serializerClass : CLASS_CHAT_COMPONENT.getClasses()) {
+          final Method[] declaredMethods = serializerClass.getDeclaredMethods();
+          final Method deserialize = Arrays.stream(declaredMethods)
+            .filter(m -> Modifier.isStatic(m.getModifiers()))
+            .filter(m -> CLASS_CHAT_COMPONENT.isAssignableFrom(m.getReturnType()))
+            .filter(m -> m.getParameterCount() == 1 && m.getParameterTypes()[0].equals(String.class))
+            .min(Comparator.comparing(Method::getName)) // prefer the #a method
```

Comment:
Sorting like this seems a little unstable, also it might not work on a Mojang mapped server